### ISSUE TITLE
[Driver] Create flags

### ISSUE THREAD

**Issue by Liryna** [dokan-dev/dokany#149]
CreateFile SL_CASE_SENSITIVE, SL_FORCE_ACCESS_CHECK, SL_OPEN_PAGING_FILE are not currently handled at all. #99

https://msdn.microsoft.com/en-us/library/windows/hardware/ff548630%28v=vs.85%29.aspx

**Comment by Liryna** [comment]
`SL_OPEN_TARGET_DIRECTORY` fixed by https://github.com/dokan-dev/dokany/pull/273

**Comment by Liryna** [comment]
`SL_CASE_SENSITIVE` is now handle on kernel side 
https://github.com/dokan-dev/dokany/commit/43645604556714cb586cf00127bb1f447ceba9f7

**Comment by Liryna** [comment]
`SL_OPEN_PAGING_FILE` Paging file is not supported

**Comment by kyanha** [comment]
Is FILE_OPEN_BY_FILE_ID properly handled?

**Comment by Liryna** [comment]
No, it is not handled at all. Kernel would need to save the ID currently provided by the user and do a find during createfile. I am not very sure there is any real life benefit and usage to implement it.
I would rather look to get ride of the fcb list and replace to a tree structure for faster lookup by fileName.

**Comment by kyanha** [comment]
Kernel shouldn't have to do a find during createfile, if it can tell createfile that it's a request to open by ID.  

(for me, it would be helpful: I'm doing a filesystem with the backend data store in PostgreSQL, and it would allow me to expose the database large object's numeric object identifier [basically, the primary key to the system table where large object data is stored] to bypass the lookup through the directory entry.)

More importantly, it's the closest thing that Windows has to an inode number.  I fear that trying to emulate inode functionality through any other means is very likely to break on future versions of Windows.

**Comment by kyanha** [comment]
Also, kernel shouldn't have to do much of anything except pass FILE_ID requests to user mode (with an extended DOKAN_OPERATIONS structure, which means it'd be at least a dokany v2.0 and more likely a v3.0 thing).  If the user code understands file IDs, it already knows how to do its own mapping.

(Then again, it would probably help if in that case the filesystem could return a reparse point on regular name lookups to reparse to the file ID.  I don't think there's any way to do reparses in dokany right now, right?)

**Comment by Liryna** [comment]
I understand your point, but from the point of view of the kernel, we need to get the FCB attached to the file for having a file context on sharing violation and sync between handles opened on the file. Then after we could do provide the file id as we do for the filename to know if user fs can find this file on his side.

**Comment by kyanha** [comment]
That's what the reparse would be for, to get the underlying ID'd file attached by the FCB.

But again, there's no way to get the reparse fsctls out to user mode, so it's (currently) pointless.
